climada/hazard/centroids/centr.py

```diff
@@ -478,44 +479,44 @@ def select_mask(self, sel_cen=None, reg_id=None, extent=None):
               (self.lat >= lat_min) & (self.lat <= lat_max)
             )
         return sel_cen
-
-    #TODO replace with nice GeoDataFrame util plot method.
-    def plot(self, axis=None, figsize=(9, 13), **kwargs):
-        """Plot centroids scatter points over earth
+    
+    def plot(self, axis=None, figsize=(9, 13), *args, **kwargs):
+        """Plot centroids geodataframe using geopandas and cartopy plotting functions.
 
         Parameters
         ----------
-        axis : matplotlib.axes._subplots.AxesSubplot, optional
-            axis to use
+        axis: optional
+            user-defined cartopy.mpl.geoaxes.GeoAxes instance
         figsize: (float, float), optional
             figure size for plt.subplots
             The default is (9, 13)
+        args : optional
+            positional arguments for geopandas.GeoDataFrame.plot
         kwargs : optional
-            arguments for scatter matplotlib function
-
+            keyword arguments for geopandas.GeoDataFrame.plot
+            
         Returns
         -------
-        axis : matplotlib.axes._subplots.AxesSubplot
+        ax : cartopy.mpl.geoaxes.GeoAxes instance
         """
-        pad = np.abs(u_coord.get_resolution(self.lat, self.lon)).min()
-
-        proj_data, _ = u_plot.get_transformation(self.crs)
-        proj_plot = proj_data
-        if isinstance(proj_data, ccrs.PlateCarree):
-            # use different projections for plot and data to shift the central lon in the plot
-            xmin, ymin, xmax, ymax = u_coord.latlon_bounds(self.lat, self.lon, buffer=pad)
-            proj_plot = ccrs.PlateCarree(central_longitude=0.5 * (xmin + xmax))
-        else:
-            xmin, ymin, xmax, ymax = (self.lon.min() - pad, self.lat.min() - pad,
-                                      self.lon.max() + pad, self.lat.max() + pad)
+        if axis == None:
+            fig, axis = plt.subplots(figsize=figsize, subplot_kw={"projection": ccrs.PlateCarree()})
```

Comment:
This is a bit inconsistent now. Here, you set the projection always to PlateCarree. Below, you use the method to_default_crs. It is then neither clear nor guaranteed that these are the same. Why not have the ccrs as a method variable, and then use `centroids.to_crs` instead?

Reply:
Well PlateCarree and DEF_CRS always work together, which is why I am always checking if Centroids.crs == DEF_CRS (and reprojecting if not) and giving PlateCarree to the transform argument (line 517). 

The line you're commenting on here is just defining that the projection the centroids are plotted on is PlateCarree. So this is then a question, whether we want to add projection as an argument to the function for the user to define. Right now the user can do that if they pass a previously defined axis to the function.

I am just not sure how it would work having both axis and projection arguments, since it would result in an error about passing multiple projection arguments... Which we could handle for sure but then this is more lines of code and I thought we wanted to keep this function simple

Reply:
good points! Maybe then we should keep it as is.

Reply:
Ok great, I'll keep it as is unless someone can think of an elegant way to do this that I haven't considered